Fix the following warnings:

  ../../components/autofill/core/browser/data_model/autofill_i18n_api.h:19:30: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  constexpr inline std::string kLegacyHierarchyCountryCodeString{"XX"};
                               ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_api.h:20:30: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  constexpr AddressCountryCode kLegacyHierarchyCountryCode =
                               ^

  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:44:32: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr Decomposition kDecompositionList[] = {
                                 ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:57:30: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr ExtractPart kExtractPartList[]{
                               ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:72:39: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr DecompositionCascade kDecompositionCascade_0 = DecompositionCascade("", kDecompositionCascade_0_Alternatives);
                                        ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:76:31: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr ExtractParts kExtractParts_0 = ExtractParts("", kExtractParts_0_Pieces);
                                ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:78:31: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr ExtractParts kExtractParts_1 = ExtractParts("", kExtractParts_1_Pieces);
                                ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:80:31: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr ExtractParts kExtractParts_2 = ExtractParts("", kExtractParts_2_Pieces);
                                ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:82:31: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr ExtractParts kExtractParts_3 = ExtractParts("", kExtractParts_3_Pieces);
                                ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:84:31: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr ExtractParts kExtractParts_4 = ExtractParts("", kExtractParts_4_Pieces);
                                ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:86:31: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr ExtractParts kExtractParts_5 = ExtractParts("", kExtractParts_5_Pieces);
                                ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:88:31: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr ExtractParts kExtractParts_6 = ExtractParts("", kExtractParts_6_Pieces);
                                ^
  ../../components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h:90:31: warning: declaration requires an exit-time destructor [-Wexit-time-destructors]
  inline constexpr ExtractParts kExtractParts_7 = ExtractParts("", kExtractParts_7_Pieces);
                                ^

--- a/components/autofill/core/browser/data_model/autofill_i18n_api.h
+++ b/components/autofill/core/browser/data_model/autofill_i18n_api.h
@@ -16,7 +16,9 @@ namespace autofill::i18n_model_definitio
 // Country code that represents autofill's legacy address hierarchy model as
 // stored `kAutofillModelRules`. As a workaround for GCC we declare the
 // std::string constexpr first.
+[[clang::no_destroy]]
 constexpr inline std::string kLegacyHierarchyCountryCodeString{"XX"};
+[[clang::no_destroy]]
 constexpr AddressCountryCode kLegacyHierarchyCountryCode =
     AddressCountryCode(kLegacyHierarchyCountryCodeString);
 
--- a/components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h
+++ b/components/autofill/core/browser/data_model/autofill_i18n_parsing_expressions.h
@@ -49,6 +49,7 @@ inline constexpr char kRegularExpression
 
 
 // Section for singular decomposition(s).
+[[clang::no_destroy]]
 inline constexpr Decomposition kDecompositionList[] = {
     Decomposition(kRegularExpression_1, true, true),
     Decomposition(kRegularExpression_2, true, true),
@@ -70,6 +71,7 @@ inline constexpr Decomposition kDecompos
 };
 
 // Section for singular extract part(s).
+[[clang::no_destroy]]
 inline constexpr ExtractPart kExtractPartList[]{
     ExtractPart("", kRegularExpression_6),
     ExtractPart("", kRegularExpression_7),
@@ -85,24 +87,33 @@ inline constexpr ExtractPart kExtractPar
 
 // Section for decomposition cascades and their alternatives.
 inline constexpr AutofillParsingProcess const* kDecompositionCascade_0_Alternatives[]{ &kDecompositionList[7], &kDecompositionList[8]};
+[[clang::no_destroy]]
 inline constexpr DecompositionCascade kDecompositionCascade_0 = DecompositionCascade("", kDecompositionCascade_0_Alternatives);
 
 // Section for extract parts and their pieces.
 inline constexpr ExtractPart const* kExtractParts_0_Pieces[]{&kExtractPartList[0],&kExtractPartList[1],&kExtractPartList[2]};
+[[clang::no_destroy]]
 inline constexpr ExtractParts kExtractParts_0 = ExtractParts("", kExtractParts_0_Pieces);
 inline constexpr ExtractPart const* kExtractParts_1_Pieces[]{&kExtractPartList[0],&kExtractPartList[1],&kExtractPartList[2]};
+[[clang::no_destroy]]
 inline constexpr ExtractParts kExtractParts_1 = ExtractParts("", kExtractParts_1_Pieces);
 inline constexpr ExtractPart const* kExtractParts_2_Pieces[]{&kExtractPartList[0],&kExtractPartList[1],&kExtractPartList[2],&kExtractPartList[3]};
+[[clang::no_destroy]]
 inline constexpr ExtractParts kExtractParts_2 = ExtractParts("", kExtractParts_2_Pieces);
 inline constexpr ExtractPart const* kExtractParts_3_Pieces[]{&kExtractPartList[4],&kExtractPartList[0],&kExtractPartList[1],&kExtractPartList[2],&kExtractPartList[3]};
+[[clang::no_destroy]]
 inline constexpr ExtractParts kExtractParts_3 = ExtractParts("", kExtractParts_3_Pieces);
 inline constexpr ExtractPart const* kExtractParts_4_Pieces[]{&kExtractPartList[5],&kExtractPartList[6]};
+[[clang::no_destroy]]
 inline constexpr ExtractParts kExtractParts_4 = ExtractParts("", kExtractParts_4_Pieces);
 inline constexpr ExtractPart const* kExtractParts_5_Pieces[]{&kExtractPartList[7],&kExtractPartList[8]};
+[[clang::no_destroy]]
 inline constexpr ExtractParts kExtractParts_5 = ExtractParts("", kExtractParts_5_Pieces);
 inline constexpr ExtractPart const* kExtractParts_6_Pieces[]{&kExtractPartList[7],&kExtractPartList[8]};
+[[clang::no_destroy]]
 inline constexpr ExtractParts kExtractParts_6 = ExtractParts("", kExtractParts_6_Pieces);
 inline constexpr ExtractPart const* kExtractParts_7_Pieces[]{&kExtractPartList[9],&kExtractPartList[5],&kExtractPartList[6],&kExtractPartList[7],&kExtractPartList[8]};
+[[clang::no_destroy]]
 inline constexpr ExtractParts kExtractParts_7 = ExtractParts("", kExtractParts_7_Pieces);
 }  // namespace
 
